onduty/index.ts

```diff
@@ -14,12 +14,46 @@
  * limitations under the License.
  */
 
+import {GitHub} from './src/github';
+import {Octokit} from '@octokit/rest';
+import {OndutyBot} from './src/bot';
+import {RotationReporterPayload, RotationTeamMap} from 'onduty';
 import express from 'express';
+import statusCodes from 'http-status-codes';
 
-module.exports.refreshRotation = (
+require('dotenv').config();
+const {
+  GITHUB_ACCESS_TOKEN,
+  GITHUB_ORG,
+  RELEASE_ONDUTY_TEAM,
+  BUILD_COP_TEAM,
+} = process.env;
+
+const ROTATION_TEAMS: RotationTeamMap = {
+  'release-on-duty': RELEASE_ONDUTY_TEAM,
+  'build-cop': BUILD_COP_TEAM,
+};
+
+const octokit = new Octokit({auth: `token ${GITHUB_ACCESS_TOKEN}`});
+const github = new GitHub(octokit, GITHUB_ORG);
+const bot = new OndutyBot(github, ROTATION_TEAMS);
+
+export async function refreshRotation(
   req: express.Request,
   res: express.Response
-) => {
-  // TODO(#778): Update GitHub teams based on rotations.
-  res.send('Triggered `reportRotation`');
-};
+) {
+  const {access_token: token, ...rotations}: RotationReporterPayload = req.body;
+
+  try {
+    if (token === GITHUB_ACCESS_TOKEN) {
+      await bot.handleUpdate(rotations);
+      res.sendStatus(statusCodes.OK);
+    } else {
+      res.sendStatus(statusCodes.UNAUTHORIZED);
+    }
+  } catch (e) {
+    console.error(e);
```

Comment:
Nit: `console.error('Some prefix explaining what happened and where', e)`
Super-nit: I started becoming a fan of `console.exception` for exceptions, even though it's [functionally the same](https://developer.mozilla.org/en-US/docs/Web/API/Console/error)

Reply:
Since this is the catch-all handler to ensure "other" errors are logged, there's not really any additional info about what happened. It's the alternative to an unhandled promise rejection. Noted re: `console.exception`, I'll try using that!